ReactWindows/ReactNative.Shared/UIManager/UIManagerModule.cs

```diff
@@ -163,12 +178,32 @@ public int AddMeasuredRootView(ReactRootView rootView)
         /// Detaches a root view from the size monitoring hooks in preparation for the unmount
         /// </summary>
         /// <param name="rootView">The root view instance.</param>
-        public void DetachRootView(ReactRootView rootView)
+        public async Task DetachRootViewAsync(ReactRootView rootView)
+        {
+            // Called on main dispatcher thread
+            DispatcherHelpers.AssertOnDispatcher();
+
+            await DispatcherHelpers.CallOnDispatcher(rootView.Dispatcher, () =>
+            {
+                rootView.RemoveSizeChanged();
+                return true;
+            }, true); // allow inlining
+        }
+
+        /// <summary>
+        /// Awaitable method completing when a root view (and a lot of dependent resources on the same dispatcher thread)
+        /// completes its cleanup.
+        /// </summary>
+        /// <param name="rootTag">The root view react tag.</param>
+        public Task CleanupRootView(int rootTag)
```

Comment:
>CleanupRootView [](start = 20, length = 15)

maybe add Async suffix since it does not actually cleanup root view before returning? Otherwise maybe rename to something like "GetRootViewCleanupTask"?